### PR TITLE
Remove ServiceReturningType, ServiceWithExplicitMethod, add helpers

### PR DESCRIPTION
**Pull request by mproch**

### CODE REVIEW

File: engine/components/sql/src/test/scala/pl/touk/nussknacker/sql/utils/BaseDatabaseQueryEnricherTest.scala [hunk 1]
```diff
@@ -2,11 +2,17 @@ package pl.touk.nussknacker.sql.utils
...
+import pl.touk.nussknacker.engine.api.context.ValidationContext
+import pl.touk.nussknacker.engine.api.context.transformation.{FailedToDefineParameter, OutputVariableNameValue}
 import pl.touk.nussknacker.engine.api.deployment.DeploymentData
 import pl.touk.nussknacker.engine.api.process.RunMode
 import pl.touk.nussknacker.engine.api.test.EmptyInvocationCollector
 import pl.touk.nussknacker.engine.api.test.InvocationCollectors.ServiceInvocationCollector
+import pl.touk.nussknacker.engine.api.typed.typing
+import pl.touk.nussknacker.sql.service.DatabaseQueryEnricher
 
+import java.lang.AssertionError
```
Comment (dswiecki) [inline engine/components/sql/src/test/scala/pl/touk/nussknacker/sql/utils/BaseDatabaseQueryEnricherTest.scala:15]: unsed

File: engine/flink/test-util/src/main/scala/pl/touk/nussknacker/engine/process/helpers/SampleNodes.scala [hunk 2]
```diff
@@ -334,30 +331,31 @@ object SampleNodes {
...
+      val returnType: typing.TypingResult = Typed.genericTypeClass[java.util.List[_]](List(listType))
+
+      EnricherContextTransformation(outputVar, returnType, new ServiceInvoker {
+        override def invokeService(params: Map[String, Any])
+                                  (implicit ec: ExecutionContext,
+                                   collector: ServiceInvocationCollector,
+                                   contextId: ContextId,
+                                   runMode: RunMode): Future[Any] = {
+            val result = (1 to count)
+              .map(line => definition.asScala.map(_ -> params("toFill").asInstanceOf[String]).toMap)
```
Comment (dswiecki) [inline engine/flink/test-util/src/main/scala/pl/touk/nussknacker/engine/process/helpers/SampleNodes.scala:351]: line can be removed

File: engine/interpreter/src/main/scala/pl/touk/nussknacker/engine/definition/MethodDefinitionExtractor.scala [hunk 3]
```diff
@@ -2,16 +2,19 @@ package pl.touk.nussknacker.engine.definition
 
 import java.lang.annotation.Annotation
 import java.lang.reflect.Method
-
+import com.github.ghik.silencer.silent
```
Comment (dswiecki) [inline engine/interpreter/src/main/scala/pl/touk/nussknacker/engine/definition/MethodDefinitionExtractor.scala:5]: we don't use silent and nowarn here, i haven't  found usage of deprecated methods here, probably can be removed

File: engine/components/openapi/src/main/scala/pl/touk/nussknacker/openapi/enrichers/BaseSwaggerEnricher.scala [hunk 4]
```diff
@@ -1,12 +1,16 @@
 package pl.touk.nussknacker.openapi.enrichers
 
+import cats.data.Validated.Valid
```
Comment (dswiecki) [inline engine/components/openapi/src/main/scala/pl/touk/nussknacker/openapi/enrichers/BaseSwaggerEnricher.scala:3]: remove

**Review by dswiecki — APPROVED** [review]
Looks much more simple now, please rebase and remove unused imports

File: engine/interpreter/src/main/scala/pl/touk/nussknacker/engine/definition/MethodDefinitionExtractor.scala (hunk 3 above)
Comment (mproch) [reply engine/interpreter/src/main/scala/pl/touk/nussknacker/engine/definition/MethodDefinitionExtractor.scala]: :+1:

File: engine/components/openapi/src/main/scala/pl/touk/nussknacker/openapi/enrichers/BaseSwaggerEnricher.scala (hunk 4 above)
Comment (mproch) [reply engine/components/openapi/src/main/scala/pl/touk/nussknacker/openapi/enrichers/BaseSwaggerEnricher.scala]: :+1:

File: engine/flink/test-util/src/main/scala/pl/touk/nussknacker/engine/process/helpers/SampleNodes.scala (hunk 2 above)
Comment (mproch) [reply engine/flink/test-util/src/main/scala/pl/touk/nussknacker/engine/process/helpers/SampleNodes.scala]: :+1:

File: engine/components/sql/src/test/scala/pl/touk/nussknacker/sql/utils/BaseDatabaseQueryEnricherTest.scala (hunk 1 above)
Comment (mproch) [reply engine/components/sql/src/test/scala/pl/touk/nussknacker/sql/utils/BaseDatabaseQueryEnricherTest.scala]: :+1:

File: engine/interpreter/src/test/scala/pl/touk/nussknacker/engine/definition/ProcessDefinitionExtractorSpec.scala [hunk 5]
```diff
@@ -52,8 +54,8 @@ class ProcessDefinitionExtractorSpec extends FunSuite with Matchers {
   test("extract definition from WithExplicitMethodToInvoke") {
     val definition = processDefinition.services("configurable1")
 
-    definition.returnType shouldBe Typed[String]
-    definition.asInstanceOf[StandardObjectWithMethodDef].methodDef.runtimeClass shouldBe classOf[Future[_]]
+    //definition.returnType shouldBe Typed[String]
```
[the remark this answers is not in the document]
Comment (mproch) [reply engine/interpreter/src/test/scala/pl/touk/nussknacker/engine/definition/ProcessDefinitionExtractorSpec.scala]: I modified the assertion

File: engine/interpreter/src/main/scala/pl/touk/nussknacker/engine/definition/MethodDefinitionExtractor.scala [hunk 6]
```diff
@@ -133,11 +129,11 @@ object MethodDefinitionExtractor {
...
         case Nil =>
           Left(s"Missing method to invoke for object: " + obj)
-        case head :: Nil =>
-          val (extractor, definition) = head
+        case head :: _ =>
+          val (_, definition) = head
           Right(definition)
-        case moreThanOne =>
-          Left(s"More than one extractor: " + moreThanOne.map(_._1) + " handles given object: " + obj)
+        //case moreThanOne =>
```
[the remark this answers is not in the document]
Comment (mproch) [reply engine/interpreter/src/main/scala/pl/touk/nussknacker/engine/definition/MethodDefinitionExtractor.scala]: I uncommented, thanks for noticing

**Pull request merged**